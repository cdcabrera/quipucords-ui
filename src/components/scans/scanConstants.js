import React from 'react';
import { translate } from '../i18n/i18n';
import { ViewToolbarFieldFilter } from '../viewToolbar/viewToolbarFieldFilter';

const ScanFilterFields = [
  /*
  {
    id: 'search_by_name',
    title: 'Name',
    placeholder: 'Filter by Name',
    filterType: 'text'
  },
  {
    id: 'search_sources_by_name',
    title: 'Source',
    placeholder: 'Filter by Source Name',
    filterType: 'text'
  }
  */
  {
    title: () => translate('toolbar.label', { context: ['option', 'search_by_name'] }),
    value: 'search_by_name',
    component: function SearchName(props) {
      return <ViewToolbarFieldFilter queryType="search_by_name" {...props} />;
    },
    selected: true
  },
  {
    title: () => translate('toolbar.label', { context: ['option', 'search_sources_by_name'] }),
    value: 'search_sources_by_name',
    component: function CredentialType(props) {
      return <ViewToolbarFieldFilter queryType="search_sources_by_name" {...props} />;
    }
  }
];

/**
 * ID: Enum with the following possible values [id, name, scan_type, most_recent_scanjob__start_time, most_recent_scanjob__status]
 */
const ScanSortFields = [
  {
    title: () => translate('toolbar.label', { context: ['option', 'name'] }),
    value: 'name',
    isNumeric: false,
    selected: true
  },
  {
    title: () => translate('toolbar.label', { context: ['option', 'most_recent'] }),
    value: 'most_recent_scanjob__start_time',
    isNumeric: true,
    sortAscending: false
  }
];

export { ScanFilterFields, ScanSortFields };
