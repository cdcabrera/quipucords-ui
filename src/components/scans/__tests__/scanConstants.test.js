import { ScanFilterFields, ScanSortFields } from '../scanConstants';

describe('ScanTypes', () => {
  it('Should have specific scanFilterFields properties', () => {
    expect(ScanFilterFields).toMatchSnapshot('scanFilterFields');
  });

  it('Should have specific scanSortFields properties', () => {
    expect(ScanSortFields).toMatchSnapshot('scanSortFields');
  });
});
