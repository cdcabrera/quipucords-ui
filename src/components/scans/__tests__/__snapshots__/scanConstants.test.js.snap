// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanTypes Should have specific scanFilterFields properties: scanFilterFields 1`] = `
Array [
  Object {
    "filterType": "text",
    "id": "search_by_name",
    "placeholder": "Filter by Name",
    "title": "Name",
  },
  Object {
    "filterType": "text",
    "id": "search_sources_by_name",
    "placeholder": "Filter by Source Name",
    "title": "Source",
  },
]
`;

exports[`ScanTypes Should have specific scanSortFields properties: scanSortFields 1`] = `
Array [
  Object {
    "id": "name",
    "isNumeric": false,
    "title": "Name",
  },
  Object {
    "id": "most_recent_scanjob__start_time",
    "isNumeric": true,
    "sortAscending": false,
    "title": "Most Recent",
  },
]
`;
